### PR TITLE
Feature/wcwpp 77 subscribe to woocommerce order state changes

### PR DESCRIPTION
**Pull request by lucapircher**
## TLDR;

- moves unit tests into a unit test directory
- updates the namespaces for the tests to match the folder structure
- updates the test pattern to match the new filenames
- introduces a settings repository and refactors existing code to read settings using the repository
- reorders the functions in the GbWeiss to be grouped by public functions, getters/setters, private functions
- disables DocComment linting rules in tests to allow adding typehints without adding a comment
- registers an action for woocommerce_order_status_changed

### CODE REVIEW

File: includes/SettingsRepository.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * Reads the plugin option with the passed name from the wordpress options
+     *
+     * @param string $name The name of the option.
+     * @return string|null
+     */
+    private function getOption(string $name): ?string
+    {
+        return \get_option(Option::OPTIONS_PREFIX . $name, null);
```
Comment (philipp-hemmer) [inline includes/SettingsRepository.php:100]: what happens if get_option returns false? i'd return null or an empty string in this case

File: tests/Integration/PluginInitializationTest.php [hunk 2]
```diff
@@ -31,9 +33,24 @@ public function test_it_does_not_pass_compatibility_check_without_woocommerce_ac
...
         $this->assertTrue(GbWeiss::checkPluginCompatabilityAndPrintErrorMessages());
     }
+
+    public function test_it_registers_an_action_for_woocommerce_order_status()
+    {
+        global $wp_filter;
+        $mock = Mockery::mock(OrderStateRepository::class);
+        $mock->shouldReceive("getAllOrderStates")->andReturn([]);
+
+        /** @var GbWeiss */
```
Comment (philipp-hemmer) [inline tests/Integration/PluginInitializationTest.php:48]: what does this comment stand for?

File: includes/SettingsRepository.php (hunk 1 above)
Comment (lucapircher) [reply includes/SettingsRepository.php:100]: Get option returns the second parameter as a default value. If it is not set false will be used, in this case I've set it to null.

File: tests/Integration/PluginInitializationTest.php (hunk 2 above)
Comment (lucapircher) [reply tests/Integration/PluginInitializationTest.php:48]: This is a typehint, It tells the php language support in the IDE that the variable is of type `GbWeiss`.

File: tests/Integration/TokenTest.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        $settingsRepository->shouldReceive("setAccessToken");
+
+        /** @var GbWeiss */
+        $plugin = GbWeiss::getInstance();
+        $plugin->setAuthenticationClient($authenticationClient);
+        $plugin->setSettingsRepository($settingsRepository);
+
+        $plugin->updateAuthToken();
+
+        $this->assertEquals('testToken', $token->getAccessToken());
```
Comment (mattcrn) [inline tests/Integration/TokenTest.php:44]: Wouldn't this assertion be true if you left out everything but the first line of this test?

Maybe assert how setAccessToken was called? 🤔

**Review by mattcrn — COMMENTED** [review]
Just a small thing in the TokenTest

File: tests/Integration/TokenTest.php (hunk 3 above)
Comment (lucapircher) [reply tests/Integration/TokenTest.php:44]: Good catch, I missed these expectations

**Review by mattcrn — APPROVED** [review]
looking good

**Review by philipp-hemmer — APPROVED** [review]

**Pull request merged**